openretina/data_io/karamanlis_2024/stimuli.py

```diff
@@ -0,0 +1,242 @@
+        f"No data directories found in {base_data_path} for animal {specie}."
+        "Please check the path and the animal species provided, and that you unrared the data files."
+    )
+
+    downsample = Resize(downsampled_size)
+
+    for session in tqdm(exp_sessions, desc="Processing sessions"):
+        session_path = os.path.join(base_data_path, session)
+        npz_file_found = False
+        mat_file = None
+        npz_file = None
+
+        # Check all files in the session path to identify relevant files
+        for recording_file in os.listdir(session_path):
+            full_path = os.path.join(session_path, recording_file)
+
+            if str(recording_file).endswith(
+                f"{stim_type}_data_extracted_stimuli_{downsampled_size[0]}_{downsampled_size[1]}.npz"
+            ):
+                npz_file = full_path
+                npz_file_found = True
+                break  # Prefer `.npz` file, no need to check further
+
+            elif str(recording_file).endswith(f"{stim_type}_data.mat"):
+                mat_file = full_path
+
+        if npz_file_found and npz_file:
+            tqdm.write(f"Loading stimuli from cached {npz_file}")
+            video_data = np.load(npz_file)
+            train_video = video_data["train_data"]
+            test_video = video_data["test_data"]
+
+        elif mat_file:
+            tqdm.write(f"Loading stimuli from {mat_file}")
+
+            train_images = load_dataset_from_h5(mat_file, "runningImages")
+            test_images = load_dataset_from_h5(mat_file, "frozenImages")
+
+            frozen_fixations = load_dataset_from_h5(mat_file, "frozenfixations").astype(int)
+            running_fixations = load_dataset_from_h5(mat_file, "runningfixations").astype(int)
+
+            test_video = return_fix_movie_torch(
+                (600, 800), rearrange(test_images, "n x y -> y x n"), frozen_fixations.T
+            )
+            test_video = downsample(rearrange(test_video, "h w n -> 1 n h w")).cpu().numpy()
+
+            train_videos = []
+            for trial in tqdm(running_fixations):
+                train_snippet = return_fix_movie_torch((600, 800), rearrange(train_images, "n x y -> y x n"), trial.T)
+                train_videos.append(downsample(rearrange(train_snippet, "h w n -> 1 n h w")))
+            train_video = torch.cat(train_videos, dim=1).cpu().numpy()
+
+            np.savez_compressed(
+                f"{mat_file[:-4]}_extracted_stimuli_{downsampled_size[0]}_{downsampled_size[1]}.npz",
+                train_data=train_video,
+                test_data=test_video,
+            )
+        else:
+            continue  # Skip session if no relevant file found
+
+        if normalize_stimuli:
+            train_video = train_video - train_video.mean() / train_video.std()
+            test_video = test_video - train_video.mean() / train_video.std()
+
+        stimuli_all_sessions["".join(session.split("/")[-1])] = MoviesTrainTestSplit(
```

Comment:
```suggestion
        stimuli_all_sessions["".join(session.split("/")[-1])] = MoviesTrainTestSplit(
```
`"".join(session.split("/")[-1])` should be the same as `session.split("/")[-1]`.